packages/playback-core/src/index.ts

```diff
@@ -1143,10 +1315,44 @@ async function handleNativeError(event: Event) {
   const { message, code } = mediaEl.error;
   const error = new MediaError(message, code);
 
+  // This accounts for cases where native playback is being used but
+  // a non-200 response occurs on the request for the playback-id's playlist.
+  // In this case, we currently already fetch the playlist in parallel (for
+  // things like inferring the stream type, live edge start window, etc.),
+  // so we'll wait briefly for that response to translate to a more accurate
+  // error.
+  if (
+    mediaEl.src &&
+    code === MediaError.MEDIA_ERR_SRC_NOT_SUPPORTED &&
+    mediaEl.readyState === HTMLMediaElement.HAVE_NOTHING
+  ) {
+    setTimeout(() => {
+      const ourError = getError(mediaEl);
+      // If the code is (still) MediaError.MEDIA_ERR_SRC_NOT_SUPPORTED,
+      // assume it's an (unlikely) case where we did, in fact, encounter
+      // media that is unsupported.
+      if (ourError?.code === MediaError.MEDIA_ERR_SRC_NOT_SUPPORTED) {
+        mediaEl.dispatchEvent(
+          new CustomEvent('error', {
+            detail: error,
+          })
+        );
+      }
+      // Since a parallel request for the source should be initiated to determine
+      // stream info (e.g. streamType) at roughly the same time as when the source
+      // is loaded by the media element, we should be able to keep this timeout short.
+      // NOTE: Although there is a case where the parallel request may happen later
+      // (namely, after metadata is loaded), this should be mutually exclusive from
+      // the case we're accounting for here, since unsupported media should not
+      // ever get metadata loaded in the first place. (CJP)
+    }, 500);
+    return;
+  }
+
   if (mediaEl.src && (code !== MediaError.MEDIA_ERR_DECODE || code !== undefined)) {
     // Attempt to get the response code from the video src url.
     try {
-      const { status } = await fetch(mediaEl.src as RequestInfo);
+      const { status } = await fetch(mediaEl.src);
```

Comment:
NOTE: I **_think_** this should no longer be necessary, since we now fetch playlists (to e.g. infer stream type) in parallel, even for native playback. Those independent requests should also now translate to errors on non-200 response cases.